main.go

```diff
@@ -117,6 +121,9 @@ func InitFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&machinePoolConcurrency, "machinepool-concurrency", 10,
 		"Number of machine pools to process simultaneously")
 
+	fs.IntVar(&clusterResourceSetConcurrency, "clusterresourceset-concurrency", 10,
```

Comment:
Is it necessary to handle concurrency for this controller? with default 10? 
it seems to me overkilling for a controller that gets triggered once in a lifetime of each cluster.

Reply:
Reduced it to 1.

Reply:
I don't see this adjusted to 1, but I'm in favor of having this be more than 1.

Reply:
Changed it back to 10 after your comment.